src/lib/utils/reverseHighlightedParts.ts

```diff
@@ -0,0 +1,39 @@
+import { HighlightedParts } from '../../types';
```

Comment:
The logic of the 3 methods are here but I did my implementation with the #2, in case we don't agree on it.

I'll remove it once the PR move out of draft.